src/middleware/__tests__/processJwt.int.test.ts

```diff
@@ -49,6 +54,44 @@ describe('processJwt', () => {
 		processJwt(mockRequest, mockResponse, nextMock);
 	});
 
+	it('does NOT populate an auth value when an auth header with an invalid issuer is sent', (done) => {
+		const mockJwt = getMockJwt({
+			iss: 'NotTheCorrectIssuer',
+		});
+		const mockRequest = {
+			headers: mockJwt,
+		} as unknown as JWTRequest;
+		const mockResponse = {} as unknown as Response;
+		const makeAssertions = async (err: unknown) => {
+			expect(err).toBeInstanceOf(Error);
+			expect(mockRequest.auth).toBe(undefined);
+		};
+		const nextMock = generateNextWithAssertions(makeAssertions, done);
+
+		processJwt(mockRequest, mockResponse, nextMock);
+	});
+
+	it('does NOT populate an auth value when an auth header with an invalid jwt server is used', (done) => {
+		const badMockJwks = getMockJwks();
```

Comment:
I realized we did NOT need to explicitly make sure this is different from the real mocked JWKS (it is, FWIW).

The reason we don't have to check is that (1) inspecting the source of the library reveals it has no cache but most importantly (2) if it *weren't* different (or if the implementation changes), then this test would fail due to the fact that there would be no error thrown by the middleware / the "bad" server would be good.

Reply:
I normally try to avoid comments as you know BUT I agree in this case this is a pretty complicated test / the research you did to enable the test would be easier with some kind of guide in this test.  I'll add it in!

Reply:
Perfect!